redux/cartSlice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const cartSlice = createSlice({
+  name: 'cart',
+  initialState: {
+    cart: <any>[],
+  },
+  reducers: {
+    addToCart: (state, action) => {
+      const itemInCart = state.cart.find((item: { id: any; }) => item.id === action.payload.id);
+      if (itemInCart) {
+        itemInCart.quantity++;
+      } else {
+        state.cart.push({ ...action.payload } as never);
+      }
+    },
+    incrementQuantity: (state, action) => {
+      const item = state.cart.find((item: { id: any; }) => item.id === action.payload);
+      item.quantity++;
+    },
+    decrementQuantity: (state, action) => {
+      const item = state.cart.find((item: { id: any; }) => item.id === action.payload);
+      if (item.quantity === 1) {
+        item.quantity = 1
```

Comment:
Good!  Saturate the quantity's lower bound at 1 and have them explicitly remove the item instead.  Makes sense.